UI/CUIControls.h

```diff
@@ -555,8 +560,11 @@ class FileDlg : public GG::FileDlg {
 public:
     /** \name Structors */ //@{
     FileDlg(const std::string& directory, const std::string& filename, bool save, bool multi,
-            const std::vector<std::pair<std::string, std::string>>& types);
+            std::vector<std::pair<std::string, std::string>> types);
```

Comment:
Can you point me to a discussion of this idiom, where by value parameters are being passed further using `std::forward`?

All my googling for `std::forward` is turning up examples where the forwarded parameter is declared as `T&&` where `T` is a template type. Here you've passed a non-template parameter by value, but are then seemingly converting it to a lvalue or rvalue reference depending how the function was called. Why isn't that information lost when calling the function and passing the parameter by value, and why isn't the result just to pass the function-scope copy of the parameter as an lvalue, regardless of the type that was passed into the function?